crates/bevy_reflect/derive/src/enum_utility.rs

```diff
@@ -1,122 +1,373 @@
+                variant_name: ::core::convert::Into::into(#variant_name),
+                field_name: ::core::convert::Into::into(#field_name)
+            })?
+        }
+    }
+
+    fn construct_field(&self, field: VariantField) -> TokenStream {
+        let bevy_reflect_path = self.reflect_enum.meta().bevy_reflect_path();
+        let alias = field.alias;
+        let field_ty = &field.field.data.ty;
+
+        quote! {
+            <#field_ty as #bevy_reflect_path::FromReflect>::from_reflect(#alias)
+                .ok_or(#bevy_reflect_path::ApplyError::MismatchedTypes {
+                    from_type: ::core::convert::Into::into(
+                        #bevy_reflect_path::DynamicTypePath::reflect_type_path(#alias)
+                    ),
+                    to_type: ::core::convert::Into::into(<#field_ty as #bevy_reflect_path::TypePath>::type_path())
+                })?
+        }
+    }
+
+    fn build(&self, this: &Ident) -> Self::Output {
+        build(self, this).expect("internal bevy_reflect error: ignored fields should be handled")
+    }
+}
+
+/// Generates the enum variant output data needed to build the `Reflect::reflect_clone` implementation.
+pub(crate) struct ReflectCloneVariantBuilder<'a> {
+    reflect_enum: &'a ReflectEnum<'a>,
+}
+
+impl<'a> ReflectCloneVariantBuilder<'a> {
+    pub fn new(reflect_enum: &'a ReflectEnum) -> Self {
+        Self { reflect_enum }
+    }
+}
+
+impl<'a> VariantBuilder for ReflectCloneVariantBuilder<'a> {
+    type Output = Option<EnumVariantOutputData>;
+    fn reflect_enum(&self) -> &ReflectEnum {
+        self.reflect_enum
+    }
+
+    fn access_field(&self, _ident: &Ident, field: VariantField) -> TokenStream {
+        let alias = field.alias;
+        quote!(#FQOption::Some(#alias))
+    }
+
+    fn unwrap_field(&self, field: VariantField) -> TokenStream {
+        let alias = field.alias;
+        quote!(#alias.unwrap())
+    }
+
+    fn construct_field(&self, field: VariantField) -> TokenStream {
+        let bevy_reflect_path = self.reflect_enum.meta().bevy_reflect_path();
+
+        let alias = field.alias;
+
+        match &field.field.attrs.clone {
+            CloneBehavior::Default => {
+                quote! {
+                    #bevy_reflect_path::Reflect::reflect_clone(#alias)?.take().ok()?
+                }
+            }
+            CloneBehavior::Trait => {
+                quote! {
+                    #FQClone::clone(#alias)
+                }
+            }
+            CloneBehavior::Func(clone_fn) => {
+                quote! {
+                    #clone_fn(#alias)
+                }
+            }
+        }
+    }
+
+    fn on_active_field(&self, _this: &Ident, field: VariantField) -> TokenStream {
+        self.construct_field(field)
+    }
+
+    fn on_ignored_field(&self, field: VariantField) -> Option<TokenStream> {
+        let alias = field.alias;
+
+        match &field.field.attrs.clone {
+            CloneBehavior::Default => None,
```

Comment:
we don't need to clone here, because the field will be cloned via `..`?
I'm a bit lost on this part

Reply:
Ah no I get it, it returns `None` so the entire `reflect_clone` returns `None`